File: abc/abc141/b.py
# https://atcoder.jp/contests/abc141/submissions/38146006

s = input()
for i in range(len(s)):
  if i % 2:
    if s[i] == 'R':
      print("No")
      break
  else:
    if s[i] == "L":
      print("No")
      break

  if i == len(s) - 1:
    print("Yes")

File: abc/abc141/b.cpp
// https://atcoder.jp/contests/abc141/submissions/27582237

#include <bits/stdc++.h>
using namespace std;
int main() {
  char c;
  string ans = "Yes";
  int i = 1;
  while(cin >> c) {
    if(i % 2 == 1) {
      if(c == 'L') ans = "No";
    } else {
      if(c == 'R') ans = "No";
    }
    i++;
  }
  cout << ans;
}
